src/main/java/tictactoe/GameSaver.java

```diff
@@ -0,0 +1,99 @@
+package tictactoe;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class GameSaver {
+
+    private final Connection connection;
+
+    public GameSaver(Connection connection) {
+        this.connection = connection;
+    }
+
+    public void saveGame(String gameName, Game game) {
+        var gameState = getGameState(gameName, game);
+        var squares = getSquares(game);
+        addEntryToDatabase(gameState, squares);
+    }
+
+    private void addEntryToDatabase
+            (ArrayList<String> gameState, String[] squares) {
+        var query = "INSERT INTO saved_games ("
+                + " game_name,"
+                + " current_player_mark,"
+                + " current_player_type,"
+                + " opponent_mark,"
+                + " opponent_type,"
+                + " squares ) VALUES ("
+                + "?, ?, ?, ?, ?, ?)";
+
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            var squaresArray = connection.createArrayOf("text", squares);
+
+            preparedStatement.setString(1, gameState.get(0));
+            preparedStatement.setString(2, gameState.get(1));
+            preparedStatement.setString(3, gameState.get(2));
+            preparedStatement.setString(4, gameState.get(3));
+            preparedStatement.setString(5, gameState.get(4));
+            preparedStatement.setArray(6, squaresArray);
+
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
+            System.out.println("Database updated successfully");
+
+        } catch (SQLException ex) {
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    public boolean isGameNameInDatabase(String gameName) {
+        var query = "SELECT * FROM saved_games WHERE game_name = ?";
+
+        try {
+            var retrievedGameName = "";
```

Comment:
Did you think about wrapping everything that is inside of `try` into a private method?

Reply:
Thanks, I'll look into this